Ops/Expr/GenXML.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace PSI;
+
+using System.Xml.Linq;
+
+// An basic XML code generator, implemented using the Visitor pattern
+public class ExprXMLGen : Visitor<XElement> {
+   XElement? mElem;

```

Comment:
Having one data member like this that is recycled and used for each call is dangerous. Suppose you set mElem, and halfway through that routine you call another Visit routine - it will get overwritten. Just make mElem a local inside each function.